QualifiedTeachersApi/src/QualifiedTeachersApi/DataStore/Crm/DataverseAdapter.CreateTeacher.cs

```diff
@@ -384,7 +389,17 @@ public async Task<CreateTeacherDuplicateTeacherResult> FindExistingTeacher()
             var result = await _dataverseAdapter._service.RetrieveMultipleAsync(query);
 
             // Old implementation returns the first record that matches on at least three attributes; replicating that here
-            var match = result.Entities.Select(entity => entity.ToEntity<Contact>()).FirstOrDefault();
+            var matches = result.Entities.Select(entity => entity.ToEntity<Contact>()).ToList();
+
+            // if a teacher exists that contains an itt record with a slugid that matches request slugid, use it
+            // in potential duplicate check
+            var teachersWithIttWithSlugs = string.IsNullOrEmpty(_command.SlugId) ? Array.Empty<Contact>() : await _dataverseAdapter.GetTeachersByInitialTeacherTrainingSlugId(_command.SlugId, columnNames: new[] { Contact.Fields.dfeta_TRN, Contact.Fields.dfeta_SlugId }, null);
+            if (teachersWithIttWithSlugs.Any())
+            {
+                matches.AddRange(teachersWithIttWithSlugs);
+            }
+
+            var match = matches.FirstOrDefault();
```

Comment:
Nit, the `// Old implementation returns...` comment should probably move above this line now